crates/rethnet/src/config/number.rs

```diff
@@ -0,0 +1,93 @@
+use rethnet_eth::U256;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq, Deserialize)]
+#[serde(untagged)]
+pub enum Number {
+    U256(U256),
+    U64(u64),
+}
+
+impl From<Number> for U256 {
+    fn from(number: Number) -> U256 {
+        match number {
+            Number::U256(number) => number,
+            Number::U64(number) => U256::from(number),
+        }
+    }
+}
+
+impl From<Number> for u64 {
+    fn from(number: Number) -> u64 {
+        match number {
+            Number::U256(number) => number
+                .try_into()
+                .expect("number is too big to fit into a u64"),
+            Number::U64(number) => number,
+        }
+    }
+}
+
+impl Serialize for Number {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match self {
+            Number::U64(number) => serializer.serialize_u64(*number),
+            Number::U256(number) => {
+                if let Ok(number) = u64::try_from(number) {
+                    serializer.serialize_u64(number)
+                } else {
+                    number.serialize(serializer)
+                }
+            }
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+#[serde(transparent)]
+pub struct NumberForU256(pub Number);
```

Comment:
I didn't notice these before, but what is their purpose? Can the same be achieved with `#[serde(with*)]`?

Reply:
I re-worked these to use `deserialize_with`, in [this commit](https://github.com/NomicFoundation/hardhat/pull/4221/commits/350bc89b0b73133e8f792c3a641ef0446c428cd5), and I'm much happier with the code that way, thank you for pushing me in that direction.